lib/MusicBrainz/Server/EditSearch/Predicate/Voter.pm

```diff
@@ -43,32 +44,65 @@ sub valid {
 sub voter_clause {
     my $self = shift;
     my $sql = 'vote.editor ';
+    my @args;
 
     if ($self->operator eq 'subscribed') {
         $sql .= 'IN (
              SELECT subscribed_editor
                FROM editor_subscribe_editor
               WHERE editor = ?
         )';
+        push @args, $self->user->id;
     } elsif ($self->operator eq 'not_subscribed') {
         $sql .= 'NOT IN (
              SELECT subscribed_editor
                FROM editor_subscribe_editor
               WHERE editor = ?
         )';
-    } elsif ($self->operator eq '!=' || $self->operator eq 'not_me') {
+        push @args, $self->user->id;
+    } elsif ($self->operator eq '=') {
+        $sql .= '= ?';
+        push @args, $self->voter_id;
+    } elsif ($self->operator eq '!=') {
         $sql .= '!= ?';
-    } else {
+        push @args, $self->voter_id;
+    } elsif ($self->operator eq 'me') {
         $sql .= '= ?';
+        push @args, $self->user->id;
+    } elsif ($self->operator eq 'not_me') {
+        $sql .= '!= ?';
+        push @args, $self->user->id;
+    } elsif ($self->operator eq 'limited') {
+        # Please keep the beginner logic in sync with Report::LimitedEditors and Entity::Editor
+        my $beginner_sql = <<~"SQL";
+            SELECT id
+              FROM editor beginner
+             WHERE id != ?
+               AND deleted = FALSE
+               AND (
+                    member_since > NOW() - INTERVAL '2 weeks'
+                    OR NOT EXISTS (
+                        SELECT 1
+                          FROM edit e2
+                         WHERE e2.editor = beginner.id
+                           AND e2.autoedit = 0
+                           AND e2.status = ?
+                        OFFSET 9
+                    )
+                )
```

Comment:
As this query is known to time out as-is, it should be limited to the last week for example.

Reply:
It doesn't feel right to have queries restricted to a hardcoded amount of time in the query itself - it seems better then to add a default time predicate to searches as you suggested at some point in the past :)

In any case, even if we hardcoded it, this wouldn't be hardcoded here, but would probably have to be some sort of more generic flag that would be then used by EditSearch::Query, maybe? So we would need to talk about how to do it more widely.